openssl/src/x509/extension.rs

```diff
@@ -506,14 +511,38 @@ impl SubjectAlternativeName {
 
     /// Sets the `otherName` flag.
     ///
-    /// Not currently actually supported, always panics.
-    #[deprecated = "other_name is deprecated and always panics. Please file a bug if you have a use case for this."]
+    /// Not currently actually supported, always panics. Please use other_name2 or other_name_ia5string
+    #[deprecated = "other_name is deprecated and always panics. Please use other_name2 or other_name_ia5string."]
     pub fn other_name(&mut self, _other_name: &str) -> &mut SubjectAlternativeName {
         unimplemented!(
-            "This has not yet been adapted for the new internals. File a bug if you need this."
+            "This has not yet been adapted for the new internals. Use other_name2 or other_name_ia5string."
         );
     }
 
+    /// Sets the `otherName` flag.
+    ///
+    /// `content` must be a valid der encoded ASN1_TYPE
+    ///
+    /// If you want to add just a ia5string use `other_name_ia5string`
+    pub fn other_name2(
+        &mut self,
+        oid: Asn1Object,
+        content: &[u8],
+    ) -> Result<&mut SubjectAlternativeName, ErrorStack> {
```

Comment:
@sfackler do you have an opinion on this returning a `Result`, vs. returning `&mut SAN` and doing the fallible ops in `build`?

Reply:
Given that we're deferring the other fallible work to build currently, I'd have this do the same I think.

Reply:
Ok, so in that case this should just return the builder, and we can conver to `ASN1_TYPE` in `build` and I think that will fix the other soundness problem

Reply:
ok, done